tests/framework/pvc.go

```diff
@@ -492,3 +429,41 @@ func addVolumeMounts(pvc *k8sv1.PersistentVolumeClaim, volumeName string) []k8sv
 	}
 	return volumeMounts
 }
+
+// GetImageInfo returns qemu-img information about given image
+func (f *Framework) GetImageInfo(namespace *k8sv1.Namespace, pvc *k8sv1.PersistentVolumeClaim) (image.ImgInfo, error) {
+	var err error
+	var info image.ImgInfo
+	executorPod := f.startVerifierPod(namespace, pvc)
+
+	cmd := fmt.Sprintf("qemu-img info %s/disk.img --output=json", utils.DefaultPvcMountPath)
+	output, stderr, err := f.ExecShellInPod(executorPod.Name, namespace.Name, cmd)
+
+	if err != nil {
+		fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: stderr: [%s]\n", stderr)
+		return info, err
+	}
+	fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: qemu-img info output %s\n", output)
+
+	err = json.Unmarshal([]byte(output), &info)
+	if err != nil {
+		klog.Errorf("Invalid JSON:\n%s\n", string(output))
+		return info, err
+	}
+
+	return info, nil
+}
+
+func (f *Framework) startVerifierPod(namespace *k8sv1.Namespace, pvc *k8sv1.PersistentVolumeClaim) *k8sv1.Pod {
+	var executorPod *k8sv1.Pod
+	var err error
+
+	executorPod, err = f.CreateVerifierPodWithPVC(namespace.Name, pvc)
+	if !apierrs.IsAlreadyExists(err) {
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	}
+	err = utils.WaitTimeoutForPodReady(f.K8sClient, executorPod.Name, namespace.Name, utils.PodWaitForTime)
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
```

Comment:
You may return (*k8sv1.Pod, error) here and  Expect() in the callers

Reply:
:+1:

Reply:
And this makes SonarCloud unhappy about duplicated code :man_facepalming: